Generic_Environment.php

```diff
@@ -257,11 +265,18 @@ private function robots_rules_generate() {
 	 * @throws Util_WpFile_FilesystemOperationException with S/FTP form if it can't get the required filesystem credentials.
 	 */
 	private function robots_rules_remove( $exs ) {
-		Util_Rule::remove_rules(
-			$exs,
-			Util_Environment::site_path() . 'robots.txt',
-			W3TC_MARKER_BEGIN_ROBOTS,
-			W3TC_MARKER_END_ROBOTS
-		);
+		$robots_path = Util_Environment::site_path() . 'robots.txt';
+
+	    Util_Rule::remove_rules(
+	        $exs,
+	        $robots_path,
+	        W3TC_MARKER_BEGIN_ROBOTS,
+	        W3TC_MARKER_END_ROBOTS
+	    );
+
+		global $wp_filesystem;
```

Comment:
Before using `$wp_filesystem`, ensure that it is first initialized with `WP_Filesystem();`.